File: scratch/fix_routes.js
const fs = require('fs');
const path = require('path');

const dirPath = path.join(__dirname, '..', 'resources', 'js');
const envVar = 'import.meta.env.VITE_ADMIN_PORTAL_PREFIX';

function walkDir(dir) {
    fs.readdirSync(dir).forEach(file => {
        let fullPath = path.join(dir, file);
        if (fs.lstatSync(fullPath).isDirectory()) {
            walkDir(fullPath);
        } else if (fullPath.endsWith('.jsx') || fullPath.endsWith('.js')) {
            processFile(fullPath);
        }
    });
}

function processFile(filepath) {
    let content = fs.readFileSync(filepath, 'utf8');
    
    if (!content.includes('/admin/')) return;
    
    let original = content;

    // 1. JSX href="/admin/..." -> href={`/${env_var}/...`}
    content = content.replace(/([a-zA-Z_]+)="\/admin\/([^"]*)"/g, `$1={\`/\${${envVar}}/$2\`}`);
    
    // 2. String literal "/admin/..." -> `/${env_var}/...`
    content = content.replace(/"\/admin\/([^"]*)"/g, `\`/\${${envVar}}/$1\``);
    
    // 3. String literal '/admin/...' -> `/${env_var}/...`
    content = content.replace(/'\/admin\/([^']*)'/g, `\`/\${${envVar}}/$1\``);
    
    // 4. Template literal `/admin/...` -> `/${env_var}/...`
    content = content.replace(/`\/admin\/([^`]*)`/g, `\`/\${${envVar}}/$1\``);
    
    if (content !== original) {
        fs.writeFileSync(filepath, content, 'utf8');
        console.log(`Updated ${filepath}`);
    }
}

walkDir(dirPath);

File: scratch/fix_routes.py
import os
import re

dir_path = r"d:\Github\tracer-study-system\resources\js"
env_var = "import.meta.env.VITE_ADMIN_PORTAL_PREFIX"

for root, dirs, files in os.walk(dir_path):
    for file in files:
        if file.endswith('.jsx') or file.endswith('.js'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()

            if '/admin/' not in content:
                continue

            original_content = content
            
            # 1. JSX href="/admin/..." -> href={`/${env_var}/...`}
            content = re.sub(r'([a-zA-Z_]+)="\/admin\/([^"]*)"', r'\1={`/${' + env_var + r'}/\2`}', content)
            
            # 2. String literal "/admin/..." -> `/${env_var}/...`
            content = re.sub(r'"\/admin\/([^"]*)"', r'`/${' + env_var + r'}/\1`', content)
            
            # 3. String literal '/admin/...' -> `/${env_var}/...`
            content = re.sub(r'\'\/admin\/([^\']*)\'', r'`/${' + env_var + r'}/\1`', content)
            
            # 4. Template literal `/admin/...` -> `/${env_var}/...`
            content = re.sub(r'`\/admin\/([^`]*)`', r'`/${' + env_var + r'}/\1`', content)
            
            if content != original_content:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(content)
                print(f"Updated {filepath}")
